src/backends/inproc_backend.cpp

```diff
@@ -86,6 +86,21 @@ static void handle_signal(int signum, siginfo_t *info, void *user_context) {
         exc.set_by_key("value", Value::new_string(sig_slot ? sig_slot->sigdesc
                                                            : "UnknownSignal"));
 
+        Value mechanism = Value::new_object();
+        Value mechanism_meta = Value::new_object();
+        Value signal_meta = Value::new_object();
+        if (sig_slot) {
+            signal_meta.set_by_key("name",
+                                   Value::new_string(sig_slot->signame));
+            signal_meta.set_by_key("number",
+                                   Value::new_int32((int32_t)sig_slot->signum));
+        }
+        mechanism_meta.set_by_key("signal", signal_meta);
+        mechanism.set_by_key("type", Value::new_string("signalhandler"));
+        mechanism.set_by_key("synthetic", Value::new_bool(true));
```

Comment:
is it a new field? we didn't map it so far.
https://docs.sentry.io/development/sdk-dev/event-payloads/exception/#exception-mechanism

Reply:
Seems like it’s missing in the docs :-/

Reply:
ok, I've added it on our end.